### DISCUSSION TITLE
[DX] <Vercel bot going rogue>

### DISCUSSION THREAD

**Discussion by ykdojo** [ykdojo/defaang#56] [Tech]
The Vercel bot started acting out and keeps asking me to authorize some stuff:

<img width="1008" alt="image" src="https://user-images.githubusercontent.com/1811651/184505420-c98d3c61-dbf3-42ba-8992-409914ca094d.png">

We should probably contact them and ask them what's going on, or I'll look into it some more.

**Comment by Dun-sin** [comment]
> The Vercel bot started acting out and keeps asking me to authorize some stuff:
> 
> <img alt="image" width="1008" src="https://user-images.githubusercontent.com/1811651/184505420-c98d3c61-dbf3-42ba-8992-409914ca094d.png">
> 
> We should probably contact them and ask them what's going on, or I'll look into it some more.

Actually it's not acting rouge, it's happening because pull request are made from people who don't have access to this repo, you are the only one who does have access to this repo and it's deployment so it's letting you know that someone else is trying to deploy something without your permission.

**Comment by mark-omarov** [comment]
question: maybe it's worth omitting deployments of PRs until they have been merged? Unless, there's a reason you'd want to deploy on every PR.

**Reply by Dun-sin** [reply]
> question: maybe it's worth omitting deployments of PRs until they have been merged? Unless, there's a reason you'd want to deploy on every PR.

We should allow previews on every pr instead.

**Reply by ykdojo** [reply]
@Dun-sin there's a way to do that?

**Reply by Dun-sin** [reply]
> @Dun-sin there's a way to do that?

Yes, I did this on my open source project using GitHub actions

**Reply by ykdojo** [reply]
Ohh interesting. @Dun-sin wanna make an issue for it?

**Reply by ykdojo** [reply]
@Dun-sin wait thinking about it again, I think it's actually ok to keep it as is.

With auto approvals, we risk people stealing user info and things - so it might not be a good idea

**Reply by ykdojo** [reply]
I thought about this again, and I guess it's okay since this is just the front end? I'm not 100% sure though

**Reply by Dun-sin** [reply]
> I thought about this again, and I guess it's okay since this is just the front end? I'm not 100% sure though

It's just vercel, so it doesn't work for backend, and it's really useful when you want to see what visual changes someone has done or test new changes without merging

**Reply by ykdojo** [reply]
@Dun-sin ok got it - wanna make a new issue for it and link this discussion?

**Reply by Dun-sin** [reply]
> @Dun-sin ok got it - wanna make a new issue for it and link this discussion?

Sure

**Reply by mark-omarov** [reply]
I suppose visual representation is nice, a nice touch would be to get regression testing or something like cypress could be interesting. I believe it comes free for OSS as well.

**Reply by ykdojo** [reply]
Update: https://github.com/ykdojo/defaang/discussions/111
